app/models/channel.rb

```diff
@@ -351,6 +351,10 @@ def update_site_banner_lookup!(skip_site_banner_info_lookup: false)
     site_banner_lookup.sync!
   end
 
+  def has_valid_uphold_connection?
+    (publisher.uphold_connection&.id == uphold_connection&.uphold_connection_id) && uphold_connection&.address && uphold_connection&.card_id
```

Comment:
Just saw this, what if they both are `nil`?
`publisher.uphold_connection&.id == uphold_connection&.uphold_connection_id`
We need to check that at least one of them exists